File: mycollab-mobile/src/main/scala/com/mycollab/mobile/module/crm/view/campaign/CampaignUrlResolver.scala
/**
 * This file is part of mycollab-mobile.
 *
 * mycollab-mobile is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-mobile is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-mobile.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.mobile.module.crm.view.campaign

import com.mycollab.common.UrlTokenizer
import com.mycollab.eventmanager.EventBusFactory
import com.mycollab.mobile.module.crm.events.{CampaignEvent, CrmEvent}
import com.mycollab.mobile.module.crm.{CrmModuleScreenData, CrmUrlResolver}
import com.mycollab.module.crm.domain.Account
import com.mycollab.module.crm.i18n.CampaignI18nEnum
import com.mycollab.vaadin.AppContext

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
class CampaignUrlResolver extends CrmUrlResolver {
  this.addSubResolver("list", new CampaignListUrlResolver)
  this.addSubResolver("add", new CampaignAddUrlResolver)
  this.addSubResolver("edit", new CampaignEditUrlResolver)
  this.addSubResolver("preview", new CampaignPreviewUrlResolver)
  
  class CampaignListUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      EventBusFactory.getInstance().post(new CrmEvent.GotoContainer(this,
        new CrmModuleScreenData.GotoModule(AppContext.getMessage(CampaignI18nEnum.LIST))))
    }
  }
  
  class CampaignAddUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      EventBusFactory.getInstance().post(new CampaignEvent.GotoAdd(this, new Account))
    }
  }
  
  class CampaignEditUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      val campaignId = UrlTokenizer(params(0)).getInt
      EventBusFactory.getInstance().post(new CampaignEvent.GotoEdit(this, campaignId))
    }
  }
  
  class CampaignPreviewUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      val campaignId = UrlTokenizer(params(0)).getInt
      EventBusFactory.getInstance().post(new CampaignEvent.GotoRead(this, campaignId))
    }
  }
  
}

File: mycollab-mobile/src/main/java/com/mycollab/mobile/module/crm/view/campaign/CampaignUrlResolver.kt
/**
 * mycollab-mobile - Parent pom providing dependency and plugin management for applications
		built with Maven
 * Copyright © ${project.inceptionYear} MyCollab (support@mycollab.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.mobile.module.crm.view.campaign

import com.mycollab.common.UrlTokenizer
import com.mycollab.vaadin.EventBusFactory
import com.mycollab.mobile.module.crm.event.CampaignEvent
import com.mycollab.mobile.module.crm.event.CrmEvent
import com.mycollab.mobile.module.crm.view.CrmModuleScreenData
import com.mycollab.mobile.module.crm.view.CrmUrlResolver
import com.mycollab.module.crm.domain.CampaignWithBLOBs

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class CampaignUrlResolver : CrmUrlResolver() {
    init {
        this.addSubResolver("list", CampaignListUrlResolver())
        this.addSubResolver("add", CampaignAddUrlResolver())
        this.addSubResolver("edit", CampaignEditUrlResolver())
        this.addSubResolver("preview", CampaignPreviewUrlResolver())
    }

    class CampaignListUrlResolver : CrmUrlResolver() {
        override fun handlePage(vararg params: String) {
            EventBusFactory.getInstance().post(CrmEvent.GotoActivitiesView(this,
                    CrmModuleScreenData.GotoModule(arrayOf())))
        }
    }

    class CampaignAddUrlResolver : CrmUrlResolver() {
        override fun handlePage(vararg params: String) =
                EventBusFactory.getInstance().post(CampaignEvent.GotoAdd(this, CampaignWithBLOBs()))
    }

    class CampaignEditUrlResolver : CrmUrlResolver() {
        override fun handlePage(vararg params: String) {
            val campaignId = UrlTokenizer(params[0]).getInt()
            EventBusFactory.getInstance().post(CampaignEvent.GotoEdit(this, campaignId))
        }
    }

    class CampaignPreviewUrlResolver : CrmUrlResolver() {
        override fun handlePage(vararg params: String) {
            val campaignId = UrlTokenizer(params[0]).getInt()
            EventBusFactory.getInstance().post(CampaignEvent.GotoRead(this, campaignId))
        }
    }
}